content/javascript/concepts/sets/sets.md

```diff
@@ -0,0 +1,90 @@
+---
+Title: 'Set'
+Description: 'Set is a special object without keys which can only store unique values.'
+Subjects:
+  - 'Web Development'
+  - 'Computer Science'
+Tags:
+  - 'Set'
+  - 'Object'
+  - 'Collections'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+`Set` is an object in JavaScript that is a collection of unique values.
+
+A set object can be created with the `Set()` constructor function.
+
+```js
+const set = new Set();
+```
+
+## Main methods
+
+- `set.add(value)` - adds a value to the set. If the value is already in the set, it doesn't do anything.
+- `set.delete(value)` - removes the given value, returns `true` if the value existed in the set, otherwise `false`.
+- `set.has(value)` - returns `true` if the value exists in the set, otherwise `false`.
+- `set.clear()` - deletes everything from the set.
+- `set.size` - returns the number of elements in the set.
+
+## Adding value to Set
+
+```js
+set.add(1); // Set {1}
+set.add('text'); // Set {1, 'text'}
+set.add(1); // still Set {1, 'text'} because 1 is already in the set
+```
+
+## Determining Whether Value is in Set
+
+```js
+console.log(set.has(1)); // Output: true
+console.log(set.has('text')); // Output: true
+console.log(set.has(5)); // Output: false
+```
+
+## Retrieving the Size of a Set
+
+```js
+console.log(set.size); // Output: 2
+```
+
+There are 2 values in `set`: `1` and `text`
```

Comment:
Preference for the variable used throughout this entry:

- Rather than just `set` which can get confusing when we are referring to a `Set` object, `mySet` is a clearer alternative name